### PR TITLE
Drop `ARCHIVE` for simple "unpack" directory

### PR DESCRIPTION
**Pull request by portante**
The goal of this work is to have the `b0.69` branch handle backups of all tar balls, archiving via the PUT API (`pbench-results-push`), and unpacking the tar ball into the INCOMING, RESULTS, and USERS trees.

The ARCHIVE tree is no longer required since there is no need to access the tar ball once it is backed up and unpacked.  The old method of using state directories is replaced with a simple "unpack" directory where the tar ball is placed in that directory when it is ready to be unpacked, and removed once it is successfully unpacked.

The removal of the ARCHIVE tree, however, has a cascading effect as it touches many files, mostly the "gold" files of the tests.

Some particular notes on various cron jobs and interfaces:

 * `pbench-dispatch`
   * Re-worked to use an unpack directory tree instead of symbolic links
     to track operational state
 * `pbench-unpack-tarballs`
   * Reworked to pull tar balls from the unpack directories
 * `pbench-audit-server`
   * This is mostly a removal of the entire report on the ARCHIVE tree, but also removed validation against the ARCHIVE
 * `pbench-check-tb-age`
   * References were for validation, dropped
 * `pbench-clean-up-dangling-results-links`
   * Dropped entirely, as this code was only partially implemented
 * `pbench-cull-unpacked-tarballs`
   * References were for validation, dropped
 * `pbench-restore-and-unpack-tarballs`
   * Switched to just using the backup tree
 * `pbench-sync-satellite`
   * No impact, did not require the ARCHIVE tree since it feeds dispatch
 * `pbench-tarball-stats`
   * Dropped entirely, similar reports can be generated by the main line code

----

Commit message body below, commit summary is PR title:

    The goal of this work is to have the `b0.69` branch handle backups of
    all tar balls, archiving via the PUT API (`pbench-results-push`), and
    unpacking the tar ball into the INCOMING, RESULTS, and USERS trees.

    The ARCHIVE tree is no longer required since there is no need to access
    the tar ball once it is backed up and unpacked.  The old method of using
    state directories is replaced with a simple "unpack" directory where the
    tar ball is placed in that directory when it is ready to be unpacked,
    and removed once it is successfully unpacked.

    The removal of the ARCHIVE tree, however, has a cascading effect as it
    touches many files, mostly the "gold" files of the tests.

    Some particular notes on various cron jobs and interfaces:

     * `pbench-dispatch`
       * Re-worked to use an unpack directory tree instead of symbolic links
         to track operational state
     * `pbench-unpack-tarballs`
       * Reworked to pull tar balls from the unpack directories
     * `pbench-audit-server`
       * This is mostly a removal of the entire report on the ARCHIVE tree,
         but also removed validation against the ARCHIVE
     * `pbench-check-tb-age`
       * References were for validation, dropped
     * `pbench-clean-up-dangling-results-links`
       * Dropped entirely, as this code was only partially implemented
     * `pbench-cull-unpacked-tarballs`
       * References were for validation, dropped
     * `pbench-restore-and-unpack-tarballs`
       * Switched to just using the backup tree
     * `pbench-sync-satellite`
       * No impact, did not require the ARCHIVE tree since it feeds dispatch
     * `pbench-tarball-stats`
       * Dropped entirely, similar reports can be generated by the main line code

### CODE REVIEW

**Comment by portante** [comment]
This does not work yet.

I still have to implement the unpack directory changes, and get the tests working again.

**Comment by portante** [comment]
Okay, this is _*mostly*_ working, where only `test-18` and `test-19` are failing, which are the tests for `pbench-restore-and-unpack-tarballs`.

**Comment by portante** [comment]
Should have passing tests now, but I need to tweak the tests a bit more to drop the old `archive` tree on-disk references in the gold files.

File: server/rpm/pbench-server.spec.j2 [hunk 1]
```diff
@@ -130,14 +130,12 @@ fi
...
 /%{installdir}/bin/pbench-check-tb-age.py
-/%{installdir}/bin/pbench-clean-up-dangling-results-links
-/%{installdir}/bin/pbench-clean-up-dangling-results-links.sh
-/%{installdir}/bin/pbench-server-config
 /%{installdir}/bin/pbench-cull-unpacked-tarballs
 /%{installdir}/bin/pbench-cull-unpacked-tarballs.py
 /%{installdir}/bin/pbench-dispatch
 /%{installdir}/bin/pbench-dispatch.sh
 /%{installdir}/bin/pbench-server-activate-create-crontab
+/%{installdir}/bin/pbench-server-config
```
Comment (dbutenhof) [inline server/rpm/pbench-server.spec.j2:138]: `pbench-restore-unpacked[.py]`??

**Review by dbutenhof — DISMISSED** [review]
I have several comments, but I don't think any of this is critical given the time constraints so I'm approving.

File: server/test/gold/test-12.txt [hunk 2]
```diff
@@ -96,17 +86,13 @@ drwxrwxr-x          - tmp
 ----- pbench-dispatch/pbench-dispatch.error
 +++++ pbench-dispatch/pbench-dispatch.log
 tarball_for-put-api_1970.01.01T00.00.00.tar.xz: OK
-run-1970-01-01T00:00:42-UTC: SAT::controller/tarball_for-put-api_1970.01.01T00.00.00: success
 tarball_for-put-api_1970.01.01T00.00.00.tar.xz: OK
-run-1970-01-01T00:00:42-UTC: controller/tarball_for-put-api_1970.01.01T00.00.00: success
-@cee:{"pbench": {"report": {"summary": {"end_ts": "1970-01-01T00:00:42-UTC", "errors": 0, "ndups": 0, "npartialsucc": 0, "nquarantined": 0, "ntbs": 2, "ntotal": 2, "prog": "pbench-dispatch", "start_ts": "1970-01-01T00:00:42-UTC", "text": "(unit-test) Processed 2 result tar balls, 2 successfully (0 partial), with 0 quarantined tar balls, 0 duplicately-named tar balls, and 0 errors."}}}}
+@cee:{"pbench": {"report": {"summary": {"end_ts": "1970-01-01T00:00:42-UTC", "errors": 0, "nberrs": 0, "nquarantined": 0, "ntbs": 0, "ntotal": 2, "nuerrs": 0, "prog": "pbench-dispatch", "start_ts": "1970-01-01T00:00:42-UTC", "text": "(unit-test) Processed 2 result tar balls, 0 successfully, with 0 quarantined, 0 backup errors, 0 unpack setup errors, and 0 archive API errors."}}}}
```
Comment (portante) [inline server/test/gold/test-12.txt:90]: `ntbs` should be `2` here.

File: server/test/gold/test-20.txt
Comment (portante) [inline server/test/gold/test-20.txt:1]: Original state tar ball still has the archive tree in it.

**Review by dbutenhof — DISMISSED** [review]
I think my open conversations are all about comments, so I'll go ahead and approve. We will want to deal with Nick's `stderr` once the new agent code is in, but we can do that later.

**Review by dbutenhof — APPROVED** [review]

File: server/test/gold/test-10.txt
[the remark this answers is not in the document]
Comment (portante) [reply server/test/gold/test-10.txt:1]: Removed.

File: server/test/gold/test-12.txt (hunk 2 above)
Comment (portante) [reply server/test/gold/test-12.txt]: Fixed.

File: server/bin/pbench-dispatch.sh [hunk 3]
```diff
@@ -243,89 +260,64 @@ while read tbmd5; do
     fi
     pbench-results-push ${tb} ${push_options}
```
[the remark this answers is not in the document]
Comment (portante) [reply server/bin/pbench-dispatch.sh:265]: We are not going to take in those changes from PR #3348 at this time.

**Review by dbutenhof — APPROVED** [review]

**Review by npalaska — APPROVED** [review]

**Pull request merged**